flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -247,7 +247,7 @@ private MemorySegment requestMemorySegment(boolean isBlocking) throws Interrupte
 				segment = availableMemorySegments.poll();
 			}
 
-			if (isBlocking && isUnavailable()) {
+			if (availableMemorySegments.isEmpty()) {
```

Comment:
Whether to remove the condition of `isBlocking` was also mentioned in [10029](https://github.com/apache/flink/pull/10029#discussion_r340033249).
I agree to remove this condition now based on the assumption that it would has trivial impact for input request side. But this change is better to cover in a separate commit in order to trace the benchmark impact if possible, and also it is out of this commit scope.

Reply:
I reserved the ```isBlocking ``` condition. Given that one local pool is never used to serve both blocking and unblocking buffer request, there is should be no side effect. This problem can be fixed in the future.